### PR TITLE
Search: automatically select product result format for sites with WooCommerce active

### PR DESCRIPTION
**Pull request by bluefuton**
#### Changes proposed in this Pull Request:
For users setting up Jetpack Search, the default result format is 'expanded'.

This PR changes the default for sites with Woo installed and active. Those sites will use the new 'product' result format instead.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

* Apply this patch to your Jetpack site before purchasing Jetpack Search.
* Install the WooCommerce plugin on your site.
* Purchase the Jetpack Search subscription for your site.
* Perform a search and ensure that you are shown the 'product' result format.
* Check that Jetpack Search setup in the Customizer shows the 'product' result format as the current selection.

Screen Shot 2021-03-09 at 15 25 11

Repeat the above steps on a new site without installing WooCommerce. The result format should be 'expanded' (the default). 

#### Proposed changelog entry for your changes:
* Search: during setup, automatically use the product result format for sites with WooCommerce active.

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php [hunk 1]
```diff
@@ -644,4 +646,18 @@ public function auto_config_excluded_post_types() {
...
 	}
+
+	/**
+	 * Automatically set result format to 'product' if WooCommerce is installed
+	 *
+	 * @since  9.6.0
+	 */
+	public function auto_config_woo_result_format() {
+		// Check if WooCommerce plugin is active (based on https://docs.woocommerce.com/document/create-a-plugin/).
+		if ( ! in_array( 'woocommerce/woocommerce.php', apply_filters( 'active_plugins', Jetpack::get_active_plugins() ), true ) ) {
```
Comment (bluefuton) [inline projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:657]: Need to check that this works on WordPress.com.

**Review by jsnmoon — DISMISSED** [review]
Works as expected on Jetpack sites with/out WooCommerce installed. Also tested on an Atomic site with WooCommerce installed.

**Review by jeherve — DISMISSED** [review]
This works well for me. 👍

**Comment by jeherve** [comment]
> Seems to be stuck on a couple of the required checks again - any way to resolve these?

Yes, we've just enabled this new changelog requirement. I've added it for you in ac37c744c662ecacc89c4543f717699538fecba9, it should be good to merge now.

For more info on the changelogger changes, see p1HpG7-bqa-p2

**Review by jeherve — DISMISSED** [review]

**Comment by bluefuton** [comment]
Thanks for the info on changelog changes @jeherve!

I've just corrected a typo in the changelog message and the PR needs another 🆗 . @jsnmoon are you able to do the honours?

**Review by jsnmoon — APPROVED** [review]

**Comment by jeherve** [comment]
> I've just corrected a typo in the changelog message

Oops, sorry about that! 🤦‍♂️

**Pull request merged**